cmd/maya-exporter/app/collector/pool/pool.go

```diff
@@ -0,0 +1,146 @@
+package pool
+
+import (
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/util"
+	zpool "github.com/openebs/maya/pkg/zpool/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// pool implements prometheus.Collector interface
+type pool struct {
+	sync.Mutex
+	metrics
+}
+
+var (
+	// runner variable is used for executing binaries
+	runner util.Runner
+)
+
+// InitVar initialize runner variable
+func InitVar() {
+	runner = util.RealRunner{}
+}
+
+// New returns new instance of pool
+func New() *pool {
+	return &pool{
+		metrics: newMetrics(),
+	}
+}
+
+// GetInitStatus run zpool binary to verify whether zpool container
+// has started.
+func (p *pool) GetInitStatus(timeout time.Duration) {
+
+	for {
+		stdout, err := zpool.Run(timeout, runner, "status")
+		if err != nil {
+			glog.Warningf("Failed to get zpool status, error: %v, pool container may be initializing,  retry after 2s", err)
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		str := string(stdout)
+		if zpool.IsNotAvailable(str) {
+			glog.Warning("No pool available, pool must be creating, retry after 3s")
+			time.Sleep(3 * time.Second)
+			continue
+		}
+		glog.Info("\n", string(stdout))
+		break
+	}
+}
+
+// collectors returns the list of the collectors
+func (p *pool) collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		p.size,
+		p.status,
+		p.usedCapacity,
+		p.freeCapacity,
+		p.usedCapacityPercent,
+		p.zpoolCommandErrorCounter,
+	}
+}
+
+// gaugeVec returns list of Gauge vectors (prometheus's type)
+// related to zpool in which values will be set.
+// NOTE: Please donot edit the order, add new metrics at the end of
+// the list
+func (p *pool) gaugeVec() []prometheus.Gauge {
+	return []prometheus.Gauge{
+		p.size,
+		p.status,
+		p.usedCapacity,
+		p.freeCapacity,
+		p.usedCapacityPercent,
+	}
+}
+
+// Describe is implementation of Describe method of prometheus.Collector
+// interface.
+func (p *pool) Describe(ch chan<- *prometheus.Desc) {
+	for _, col := range p.collectors() {
+		col.Describe(ch)
+	}
+}
+
+func (p *pool) get() (zpool.Stats, error) {
+	p.Lock()
+	defer p.Unlock()
+	var (
+		err         error
+		stdoutZpool []byte
+		timeout     = 5 * time.Second
+		zpoolStats  = zpool.Stats{}
+	)
+
+	glog.V(2).Info("Run zpool list command")
```

Comment:
what is the log level? does this gets logged very frequently?

Reply:
Log level set to 2, so by default it will only log in case of errors.

Reply:
No, log level set to 2, so if we run exporter with `-v 2` then it will log this.

Reply:
what is value for '-v' that we use for running exporter?

Reply:
0